修正: プロパティーウィンドウのサイズが前回の設定に影響される問題を解決

asaday opened the pull request:
## 概要
#1013 で報告された、ボイスチェンジャーのプロパティーを開いた後に他のソースのプロパティーを開くと、ウィンドウサイズが前回のまま(大きいまま)になってしまう問題を修正しました。

## 問題の原因
子ウィンドウ(childWindow)は閉じずに再利用される設計ですが、前回設定された setMinimumSize() の制約が残ったままになっていました。そのため、新しい setBounds() で小さいサイズを設定しようとしても、古い最小サイズ制約により拒否されていました。

## 修正内容
- 子ウィンドウ再表示時に setMinimumSize(0, 0) で前回の最小サイズ制約をリセット
- setBounds() でサイズと位置を設定した後に、新しい最小サイズを適用

## 影響範囲
- プロパティーウィンドウなど、子ウィンドウを再利用する全ての画面
- ウィンドウサイズの設定処理

## テスト確認項目
- [x] ボイスチェンジャーのプロパティーを開いて閉じる
- [x] その後、他のソース(映像キャプチャデバイスなど)のプロパティーを開く
- [x] ウィンドウサイズが適切なサイズで表示されることを確認

Fixes #1013

asaday committed:
修正: プロパティーウィンドウのサイズが前回の最小サイズ制約により変更できない問題を解決

- 子ウィンドウ再表示時に setMinimumSize(0, 0) で前回の制約をリセット
- setBounds() でサイズと位置を設定した後に新しい最小サイズを適用
- restore() を最初に実行し、show() を最後に実行するよう処理順序を最適化

Fixes #1013

asaday committed:
修正: 子ウィンドウを閉じる際にウィンドウ状態をリセットすることで、再利用時のサイズ設定問題を解決

- window-closeChildWindow で hide() する前に setMinimumSize(0, 0) と setResizable(true) を呼び出し
- 前回の最小サイズ制約とリサイズ可否設定が次回の再利用時に影響しないよう確実にクリーンアップ
- これにより、異なるサイズのプロパティーウィンドウを開く際に正しいサイズで表示されるようになる

Fixes #1013

koizuka commented:
* ボイスチェンジャーのプロパティを閉じてから他のソースプロパティを開いてドラッグ -> サイズok
* ボイスチェンジャーのプロパティを閉じずにから他のソースプロパティを開いてドラッグ -> 動かした途端にボイスチェンジャーソースサイズに拡大した

なので、閉じるときだけでなく、閉じずに再利用するパスでも通るところで再設定が必要そうかな

asaday commented:
そこの上に元々あるコメント
      // For some unknown reason, electron sometimes gets into a
      // weird state where this will always fail.  Instead, we
      // should recover by simply setting the size and forgetting
      // about the bounds.
      
でも問題だったようで、electronではタイミングが遅延しているのか正確に反映させるのは辛いようです。
本当にちゃんとやるなら、きっちりcloseしてから処理したほうがよさそうですが、それはそれで結構遅くなりそうなのでメリットに見合うか微妙なところです。

koizuka commented:
あー、steramlabsのmain.jsみたら、 `window-showChildWindow` イベントハンドラ自体がなくなってる。なんか作り直してそうだね

asaday committed:
変更: 子ウィンドウの最小サイズ設定を遅延実行し、closeイベントでウィンドウ状態をリセット

- setMinimumSize() を setTimeout で10ms遅延させ、setBounds 適用後に確実に設定
- childWindow の close イベントハンドラーでウィンドウ状態をリセット
- window-closeChildWindow からはリセット処理を削除（close イベントで対応）
- これにより、再利用時のウィンドウサイズが正しく適用されるようになる

Fixes #1013

asaday committed:
Merge branch 'n-air_development' into fix/issue-1013-window-size-reset
